### ISSUE TITLE
 Running slowly on Android

### ISSUE THREAD

**Issue by ColdStar007** [wix/react-native-zss-rich-text-editor#154]
It works fine on iOS, but it runs very slowly on Android.

**Comment by minhphung210** [comment]
same issue

**Comment by antony-meunier** [comment]
Same here !

**Comment by arneson** [comment]
@ColdStar007 @minhphung210 @antony-meunier 
There is an infinite loop running in the android version, it has been resolved in a few different forks but there is little to no activity in this repo unfortunately. 

You can depend on the tag 1.1.1 in my fork of this repo which has a fix. 

https://github.com/arneson/react-native-zss-rich-text-editor.git#1.1.1

**Comment by minhphung210** [comment]
@arneson arn
your solution not work for me :(
thank for reply

**Comment by arneson** [comment]
That's unfortunate. Have you made sure to do a clean native build before testing? As the problem is in the editor.html file in the library which is only copied over on build, not reload.

**Comment by export-mike** [comment]
@arneson keen to make a hard fork? I'm having issues. lets collaborate I have a fork too: https://github.com/export-mike/react-native-zss-rich-text-editor

I'm keen to change this to use the recommended react-native-web-view which is faster for ios and also has window.postMessage working without hacks.

**Comment by export-mike** [comment]
@arneson where is the change to remove the infinite loop? https://github.com/arneson/react-native-zss-rich-text-editor/commits/master/src/editor.html

**Comment by arneson** [comment]
@export-mike 

Yeah, I'd love to get this library working properly again. Would love to collaborate. 

It's the commit with message fix getCaretPosition. 

https://github.com/arneson/react-native-zss-rich-text-editor/commit/7704a0efdce8f0a41f26526c5a4558a6a69c735b#diff-5764eca5893a91bd5fb61afbae89595d

Worth noting is that I did not write that commit but pulled it from charpeni 's fork.

**Comment by export-mike** [comment]
ok cool sounds great.

basically the fork of the editor I'm using has the ability to render custom buttons. I need to change it to use the react-native-web-view

I can pull in this infinite loop fix manually. what other changes have you introduced?

what functionality are you working to support/fix?

**Comment by arneson** [comment]
Working on supporting the toolbar which behaves like a fucking tool in the current version. 


But totally agree we need a new hard fork.
I can create a fork and add you as a collab, that sound good?

**Comment by export-mike** [comment]
Yeah great. basically the work I have done so far allows custom components to be rendered.

- Need to successfully remove the title editor from the editor if not required.
- Show and hide toolbar successfully on android
- Android Performance
- iOS shows a black line page-break kinda thing when scrolling.

**Comment by export-mike** [comment]
Bottom 2 points need to be completed.

**Comment by arneson** [comment]
@export-mike 
Don't think I've seen the last one. 
Could you send over a screen shot?

**Comment by export-mike** [comment]
@arneson https://join.slack.com/t/reactnativezs-ash8421/shared_invite/enQtNTA1NTMxNzE4NDY3LWM3MzQwM2ZkZTEwZjRhMGYyMDBmZTY2ZmMyYmY1ZGU0NGE4OGVjN2M4NTY1MDkyMzFmNzAxMDZlMmU2YmJlYTc

Let's move to slack

**Comment by arneson** [comment]
Saw that link way too late. It’s dead now :) @export-mike

**Comment by export-mike** [comment]
@arneson https://join.slack.com/t/reactnativezs-ash8421/shared_invite/enQtNTA1NTMxNzE4NDY3LWM3MzQwM2ZkZTEwZjRhMGYyMDBmZTY2ZmMyYmY1ZGU0NGE4OGVjN2M4NTY1MDkyMzFmNzAxMDZlMmU2YmJlYTc

**Comment by ethanyuwang** [comment]
@arneson Thanks your fork worked for me!

**Comment by ashokkumar88** [comment]
@arneson your version supports hashtag and mentions?

**Comment by arneson** [comment]
@ashokkumar88 
I have not added that functionality, so if it exists in original package then yes, otherwise no.

**Comment by export-mike** [comment]
looks like UIWebView has been deprecated. I'll be updating https://github.com/export-mike/react-native-zss-rich-text-editor to use WKWebView.
